torch/_functorch/_aot_autograd/schemas.py

```diff
@@ -194,7 +197,44 @@ class SubclassCreationMeta:
     # Used at runtime to determine the subclass type, so we don't need to save the original subclass
     original_subclass_type: Optional[type] = None
 
-    def creation_fn(self, all_args, *, is_runtime: bool):
+    def compute_outer_size_and_stride(
+        self,
+        all_args,
+        *,
+        curr_start_idx: int,
+        is_runtime: bool,
+        is_nested: bool,
+    ):
+        from .subclass_utils import compute_symint_placeholders
+
+        def compute(outer, start_idx):
+            placeholders = compute_symint_placeholders(outer)
+            has_symbolic = any(placeholders)
+
+            if is_runtime and has_symbolic:
+                if is_nested:
+                    return None, curr_start_idx
+                else:
+                    start = curr_start_idx
+                    end = start_idx + sum(placeholders)
+                    it_args = iter(all_args[start:end])
+                    it_placeholders = iter(placeholders)
+                    return pytree.tree_map_only(
+                        lambda _: next(it_placeholders), lambda _: next(it_args), outer
+                    ), start + len(placeholders)
+            return outer, start_idx
+
+        outer_size, next_idx = compute(self.outer_size, curr_start_idx)
+        outer_stride, _ = compute(self.outer_stride, next_idx)
+        return outer_size, outer_stride
+
+    def creation_fn(
+        self,
+        all_args,
+        *,
+        is_runtime: bool,
+        is_nested: bool,
```

Comment:
is it necessary for `creation_fn` to take in an `is_nested` flag? Or can we figure out a way to write it that handles the recursion naturally without needing to branch on whether we are in a nested call.

Reply:
synced offline, this is gonna be wrong for nested subclasses that both need their outer_size/stride populated